sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2ScanExec.scala

```diff
@@ -46,34 +48,46 @@ case class DataSourceV2ScanExec(
       new DataSourcePartitioning(
         s.outputPartitioning(), AttributeMap(output.map(a => a -> a.name)))
 
+    case _ if readerFactories.size == 1 => SinglePartition
```

Comment:
shall we move it before `case s: SupportsReportPartitioning`? the difference is, shall we always report `SinglePartition` if there is only one writer factory. By looking at `EnsureRequirements`, seems `SinglePartition` is more likely to satisfy the distribution.

Reply:
Makes sense to me; also added a comment to the API describing this behavior